Allow browsers to block mixed content requests from cross-origin tests.

rniwa opened the pull request:
Gecko and Blink by default block mixed content requests.

As a result, header-referrer-strict-origin-when-cross-origin.https.html, header-referrer-strict-origin.https.html,
and beacon/headers/header-referrer-unsafe-url.https.html fail in those two browser engines.

Allow the mixed content beacon requests to be blocked in these tests as allowed by the specification:
https://www.w3.org/TR/mixed-content/#should-block-fetch
https://www.w3.org/TR/mixed-content/#requirements-user-controls

Also fixed a bug in header-referrer-strict-origin-when-cross-origin.https.html that it was not testing cross-origin.

toddreifsteck commented on beacon/headers/header-referrer.js:
```diff
@@ -3,15 +3,18 @@ var RESOURCES_DIR = "/beacon/resources/";
-    assert_true(navigator.sendBeacon(testUrl), "SendBeacon Succeeded");
+    const sentBeacon = navigator.sendBeacon(testUrl);
+    assert_true(sentBeacon || mayBeBlockedAsMixedContent, "SendBeacon Succeeded");
+    if (!sentBeacon && mayBeBlockedAsMixedContent)
```
I think this would be easier to read if the pass/fail logic wasn't separated into the assert and the if (!sentBeacon)

rniwa committed:
Allow browsers to block mixed content requests from cross-origin tests.

Gecko and Blink by default block mixed content requests.

As a result, header-referrer-strict-origin-when-cross-origin.https.html, header-referrer-strict-origin.https.html,
and beacon/headers/header-referrer-unsafe-url.https.html fail in those two browser engines.

Allow the mixed content beacon requests to be blocked in these tests as allowed by the specification:
https://www.w3.org/TR/mixed-content/#should-block-fetch
https://www.w3.org/TR/mixed-content/#requirements-user-controls

Also fixed a bug in header-referrer-strict-origin-when-cross-origin.https.html that it was not testing cross-origin.
